### PR TITLE
Open Research: Fix broken links

### LINKED ISSUE #1125: [BUG] Links in Open Research chapter seem to be broken
### Bug description

* The Open Research chapter's [index ](https://the-turing-way.netlify.app/open_research/open_research.html) does not link correctly to each section/subsection
* A more specific issue seems to be around "2. Barriers to data sharing". Although the index (and the section) says there are two, only one is mentioned [in the section](https://the-turing-way.netlify.app/open_research/01/opendata.html#Barriers-to-data-sharing).

#### Expected behaviour

* Links in the [index ](https://the-turing-way.netlify.app/open_research/open_research.html) should take you to each section/subsection.
* [2. Barriers to data sharing](https://the-turing-way.netlify.app/open_research/01/opendata.html#Barriers-to-data-sharing) should list both privacy _and_ "Nationally and commercially sensitive data"

#### Actual behaviour

* Links in the [index ](https://the-turing-way.netlify.app/open_research/open_research.html) to subsections (for example: 3.1.1. Step 1: Make your data available) lead to higher level sections (here: 3. Open Data)
* I cannot read 3.2.2.National and commercially sensitive data

### PR DESCRIPTION
**Pull request by Sylfrena**
### Summary

Fix broken links for sections 3.1.2, 4.5, and 12 in the Open Research
chapter.

Fixes #1125 

### List of changes proposed in this PR (pull-request)

* *Add correct link for section 3.2.2, [National and commercially sensitive data](https://the-turing-way.netlify.app/rdm/sharingarchiving.html#national-and-commercially-sensitive-data).*
* *Add correct link for section 4.5, [How to contribute to other’s open source software projects](https://the-turing-way.netlify.app/open_research/02/opensourcesoftware.html#how-to-contribute-to-other-s-open-source-software-projects).*
* *Add correct link for section 12, [Definitions/glossary](https://the-turing-way.netlify.app/open_research/07/resources#definitions-glossary).*

### What should a reviewer concentrate their feedback on?

- [ ] *Lorem ipsum dolor sit amet, consectetur adipiscing.*
- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by Sylfrena** [comment]
Hey @Ismael-KG,

I have made changes to the links in the index. However, section 3.1.1. Step 1- Make your data available seems to work correctly for me(as mentioned in the bug).

Please take a look and let me know if I have overlooked something. Thanks!

File: book/website/open_research/open_research.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Recommended skill level: low.
         4. [Step 4: Make your data citeable](/open_research/01/opendata#step-4-make-your-data-citeable)
     2. [Barriers to data sharing](/open_research/01/opendata#barriers-to-data-sharing)
         1. [Privacy](/open_research/01/opendata#privacy)
-        2. [National and commercially sensitive data](/open_research/01/opendata#national-and-commercially-sensitive-data)
+        2. [National and commercially sensitive data](/rdm/sharingarchiving.html#national-and-commercially-sensitive-data)
```
Comment (alexmorley) [inline book/website/open_research/open_research.md:23]: :green_circle:

File: book/website/open_research/01/opendata.md [hunk 2]
```diff
@@ -98,5 +98,4 @@ Preserving privacy may still be possible if partial or generalised datasets are
 For example, providing age bands instead of birth date or only the first two digits of postal codes.
 It may also be possible to provide the data in such a format that researchers can query it whist keeping the data itself closed, for example, a user may be able to send a query to retrieve the mean of a dataset, but not be able to access any of the individual datapoints.
 
-Data privacy is also maintained when dealing with data of national or commercial interest, details for which are listed in the chapter [Research Data Management](/rdm/rdm), in the section: *National and commercially sensitive data*.
-[rdm]: /rdm
+Data privacy is also maintained when dealing with data of national or commercial interest, details for which are listed in the chapter [Research Data Management](/rdm/rdm), in the section: [*National and commercially sensitive data*](/rdm/sharingarchiving.html#national-and-commercially-sensitive-data).
```
Comment (alexmorley) [inline book/website/open_research/01/opendata.md:101]: :heavy_check_mark:

**Comment by malvikasharan** [comment]
Thank you @Sylfrena and @alexmorley for such a rapid editing and reviewing. 🌺🎉

**Comment by Sylfrena** [comment]
@alexmorley Thanks for such a quick review!

**Pull request merged**